src/core/types/ui_settings.ts

```diff
@@ -50,7 +50,7 @@ export interface UiSettingsParams<T = unknown> {
   /** used to group the configured setting in the UI */
   category?: string[];
   /** array of permitted values for this setting */
-  options?: string[];
+  options?: string[] | number[];
```

Comment:
@TinaHeiligers only FYI: this may create conflicts with https://github.com/elastic/kibana/pull/136354